Fix/max col width config

koiker opened the pull request:
This commit will fix the default/max col width that lead to different outputs because the default column width is 80 (increased the default column width to 120)
This change has included a new Config class where the program can use config defaults and override them by environment variables. In this initial commit we have the `MAX_COL_WIDTH` environment variable that when set will replace the default 120 characters from the program.
This new class incorporate pythonic code that is only compatible with Python 3.5+ and this change will break backward compatibility with Python 2.7 that has been deprecated for some time.
Another small change is the yaml loader to use `yaml.SafeLoader` as a safer Class and remove the security vulnerabilities found in `yaml.Loader` and `yaml.FullLoader`
By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

Commit:
Added config class to make more flexible some configurations.
Added Yaml max width to change the yaml output when converting data with more than 80 columns. The default is set to 120
Removed Python 2.7 compatibility due to config class type hints and other features only found in python 3.x

Commit:
Implemented max_col_width to generate yaml strings without adding extra newlines due to the default yaml max_width.

Commit:
Updated yaml.loader to the more safer yaml.FullLoader

Commit:
Removed support for Python 2.7 from this version

Commit:
Changed `yaml.FullLoader` to `yaml.SafeLoader`

stilvoid commented on cfn_tools/_config.py:
```diff
@@ -0,0 +1,128 @@
+    default: Any = None
+
+
+_CONFIG_DEFAULTS: Dict[str, _ConfigArg] = {
```
Docs need to reflect the new environment variable

stilvoid reviewed (changes requested):
Code coverage needs fixing (cover new config.py) but otherwise lgtm :)

Commit:
Changed `yaml.FullLoader` to `yaml.SafeLoader` in test_flip.py
Updated `README.md` with the new `max_col_width` feature
Added `test_config.py` for unit-tests

Commit:
Changed `cfn_clean/__init__.py` to import `collections.abc` only. (Compatible only with py3.3+)
This change will keep the test coverage fully covered.

Commit:
Changed `cfn_flip/__init__.py` to remove py27 backward compatibility

Commit:
Added Python 3.9 to the scope of compatibility test

Commit:
new _config.py coverage to 100%

Commit:
Update travis to include Python 3.9 compatibility test.

koiker commented:
Make the changes:
- Fixed coverage gap due to the Python 2.7 deprecation
- Added tests for full coverage unit-test on the new config module
- Added Python 3.9 test in tox and travis
- Added the new feature in the README.md

stilvoid committed:
Merge branch 'master' into fix/max-col-width-config
